Queue.cpp

```diff
@@ -11,6 +12,15 @@ namespace PolyM {
 
 class Queue::Impl
 {
+    struct Request
+    {
+        Request() : _available(false) {};
+        
+        std::unique_ptr<Msg> _res;
+        std::condition_variable _condVar;
+        std::atomic<bool> _available;
+    };
```

Comment:
You always need a mutex to go with a condition_variable. Please see https://en.cppreference.com/w/cpp/thread/condition_variable

There's no need to have the separate "available" member, as the response pointer being non-null already conveys the same information and can be waited on with the condition_variable.

Starting the name of the members with "_" is for private members, you can omit it with public ones.

Reply:
Yes, a mutex is always associated with a condition variable. Please see my previous comment regarding this.

Agreeing on _available that is unnecessary here.